python/pyspark/tests.py

```diff
@@ -614,6 +614,18 @@ def context_barrier(x):
         times = rdd.barrier().mapPartitions(f).map(context_barrier).collect()
         self.assertTrue(max(times) - min(times) < 1)
 
+    def test_barrier_with_python_worker_reuse(self):
+        """
+        Verify that BarrierTaskContext.barrier() with reused python worker.
+        """
+        rdd = self.sc.parallelize(range(4), 4)
```

Comment:
Let's explicitly set `spark.python.worker.reuse` or at least let's assert.

Reply:
Thanks, done in 02555b8.